### PR TITLE
[RHOBS-1009/ACM-8509]: Add ServiceMonitor for metrics-collector and instrument it properly

### PR DESCRIPTION
**Pull request by saswatamcode**
This PR adds a ServiceMonitor for metrics-collector (both standard and uwl) and instruments both the federation and remote writing.

Not adding alerts directly, as we might want to test those first to see if it is flaky.

### CODE REVIEW

File: collectors/metrics/pkg/forwarder/forwarder.go [hunk 1]
```diff
@@ -219,9 +223,21 @@ func NewWorkerMetrics(reg *prometheus.Registry) *workerMetrics {
 			Name: "federate_filtered_samples",
 			Help: "Tracks the number of samples filtered per federation",
 		}),
-		gaugeFederateErrors: promauto.With(reg).NewGauge(prometheus.GaugeOpts{
-			Name: "federate_errors",
-			Help: "The number of times forwarding federated metrics has failed",
+		federateRequests: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Name: "federate_requests_total",
```
Comment (philipgough) [inline collectors/metrics/pkg/forwarder/forwarder.go:227]: I assume there is nothing (alert, dashboard) that could break with this change of name?

Comment (saswatamcode) [reply collectors/metrics/pkg/forwarder/forwarder.go:227]: Nothing I could find in this repo at least.

File: collectors/metrics/pkg/forwarder/forwarder.go [hunk 2]
```diff
@@ -439,11 +458,13 @@ func (w *Worker) forward(ctx context.Context) error {
 
 	req := &http.Request{Method: "POST", URL: w.to}
 	err = w.toClient.RemoteWrite(ctx, req, families, w.interval)
+	w.metrics.forwardRemoteWriteRequests.Inc()
 	if err != nil {
 		statusErr := w.status.UpdateStatus("Degraded", "Failed to send metrics")
 		if statusErr != nil {
 			rlogger.Log(w.logger, rlogger.Warn, "msg", failedStatusReportMsg, "err", statusErr)
 		}
+		w.metrics.forwardRemoteWriteErrors.Inc()
```
Comment (philipgough) [inline collectors/metrics/pkg/forwarder/forwarder.go:467]: do you think its worth adding the status code perhaps to this and the other metrics?

Comment (saswatamcode) [reply collectors/metrics/pkg/forwarder/forwarder.go:467]: Will try!

**Review by philipgough — APPROVED** [review]
/lgtm

**Comment by saswatamcode** [comment]
closing this for feature branch merge

**Pull request closed without merging**